resources/qml/PrinterSelector/MachineListButton.qml

```diff
@@ -7,81 +7,134 @@ import QtQuick.Controls 2.3
 import UM 1.5 as UM
 import Cura 1.0 as Cura
 
-
-Button
-{
-    id: machineListButton
-
+Loader {
+    id: loader
     width: parent.width
-    height: UM.Theme.getSize("large_button").height
-    leftPadding: UM.Theme.getSize("default_margin").width
-    rightPadding: UM.Theme.getSize("default_margin").width
-    checkable: true
-    hoverEnabled: true
+    height: childrenRect.height
+    sourceComponent: {
+        switch (model.listType) {
+            case "HIDE_BUTTON":
+                hideButtonComponent
+                break;
+            case "SHOW_BUTTON":
+                showButtonComponent
+                break;
+            case "MACHINE":
+                machineListButtonComponent
+                break;
+            default:
+        }
+    }
+    property var onClicked
```

Comment:
Where is this property actually set? Doesn't the `OnClicked` in the `MachineListButton` (in `MachineSelectorList`) already take care of the clicks?

Reply:
It needs to be declared in MachineListButton first so that it can be referenced in MachineListButton. If you take it out when this component is loaded OnClicked will be pointing to nothing in MachineListButton before the assignment in MachineSelectorList (causing the sad red text :cry: ).